ts/Dashboards/EditMode/SidebarPopup.ts

```diff
@@ -431,12 +431,39 @@ class SidebarPopup extends BaseForm {
                             const newCell =
                                 components[i].onDrop(sidebar, dropContext);
 
-                            if (newCell) {
-                                dropContext.setHighlight();
-                                sidebar.editMode.setEditCellContext(newCell);
-                                sidebar.show(newCell);
-                                newCell.setHighlight();
-                            }
+                            addEvent(
+                                this.editMode,
+                                'layoutChanged',
+                                (e): void => {
+                                    if (newCell && e.type === 'newComponent') {
```

Comment:
Is this event removed somewhere later? I see that the listener is added every time the `drag` is started.

Reply:
no, i've added that